src/main/java/net/sf/jabref/logic/util/version/Version.java

```diff
@@ -0,0 +1,129 @@
+/*  Copyright (C) 2016 JabRef contributors.
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+package net.sf.jabref.logic.util.version;
+
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+import net.sf.jabref.logic.util.BuildInfo;
+
+/**
+ * Represents the Application Version with the major and minor number, the full Version String and if it's a developer version
+ */
+public class Version {
+
+    private final String fullVersion;
+    private final int major;
+    private final int minor;
+    private final int patch;
+    private final boolean devVersion;
+
+    /**
+     * @param version must be in form of X.X (eg 3.3; 3.4dev)
+     */
+    public Version(String version) {
+        if (version == null || "".equals(version) || version.equals(BuildInfo.UNKOWN_VERSION)) {
+            this.fullVersion = BuildInfo.UNKOWN_VERSION;
+            this.major = this.minor = this.patch = 0;
+            this.devVersion = false;
+            return;
+        }
+
+        this.fullVersion = version;
+        devVersion = version.contains("dev");
+        String[] versionParts = version.split("dev");
+        String[] versionNumbers = versionParts[0].split(Pattern.quote("."));
+        this.major = Integer.parseInt(versionNumbers[0]);
+        this.minor = versionNumbers.length >= 2 ? Integer.parseInt(versionNumbers[1]) : 0;
+        this.patch = versionNumbers.length >= 3 ? Integer.parseInt(versionNumbers[2]) : 0;
+    }
+
+    /**
+     * @return true if this version is newer than the passed one
+     */
+    /* Ignoring the SimplifyBooleanReturns request from PMD, you could write it all in one line but then it's unreadable */
+    @SuppressWarnings("PMD.SimplifyBooleanReturns")
+    public boolean isNewerThan(Version otherVersion) {
```

Comment:
Can one override `<=` for custom classes in java? If yes then I would prefer that.